_includes/tutorials/joining-stream-table/kstreams/code/src/main/java/io/confluent/developer/JoinStreamToTable.java

```diff
@@ -62,83 +51,74 @@ public Topology buildTopology(Properties envProps) {
 
         topics.add(new NewTopic(
-                envProps.getProperty("rating.topic.name"),
-                Integer.parseInt(envProps.getProperty("rating.topic.partitions")),
-                Short.parseShort(envProps.getProperty("rating.topic.replication.factor"))));
+                allProps.getProperty("rating.topic.name"),
+                Integer.parseInt(allProps.getProperty("rating.topic.partitions")),
+                Short.parseShort(allProps.getProperty("rating.topic.replication.factor"))));
 
         topics.add(new NewTopic(
-                envProps.getProperty("rated.movies.topic.name"),
-                Integer.parseInt(envProps.getProperty("rated.movies.topic.partitions")),
-                Short.parseShort(envProps.getProperty("rated.movies.topic.replication.factor"))));
+                allProps.getProperty("rated.movies.topic.name"),
+                Integer.parseInt(allProps.getProperty("rated.movies.topic.partitions")),
+                Short.parseShort(allProps.getProperty("rated.movies.topic.replication.factor"))));
 
         client.createTopics(topics);
         client.close();
     }
 
     public Properties loadEnvProperties(String fileName) throws IOException {
-        Properties envProps = new Properties();
+        Properties allProps = new Properties();
         FileInputStream input = new FileInputStream(fileName);
-        envProps.load(input);
+        allProps.load(input);
         input.close();
 
-        return envProps;
+        return allProps;
     }
 
     public static void main(String[] args) throws Exception {
-
         if (args.length < 1) {
             throw new IllegalArgumentException("This program takes one argument: the path to an environment configuration file.");
         }
 
         JoinStreamToTable ts = new JoinStreamToTable();
-        Properties envProps = ts.loadEnvProperties(args[0]);
-        Properties streamProps = ts.buildStreamsProperties(envProps);
-        Topology topology = ts.buildTopology(envProps);
+        Properties allProps = ts.loadEnvProperties(args[0]);
+        allProps.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        allProps.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
+        Topology topology = ts.buildTopology(allProps);
 
-        ts.createTopics(envProps);
+        ts.createTopics(allProps);
 
-        final KafkaStreams streams = new KafkaStreams(topology, streamProps);
+        final KafkaStreams streams = new KafkaStreams(topology, allProps);
         final CountDownLatch latch = new CountDownLatch(1);
 
         // Attach shutdown handler to catch Control-C.
         Runtime.getRuntime().addShutdownHook(new Thread("streams-shutdown-hook") {
             @Override
             public void run() {
-                streams.close();
+                streams.close(Duration.ofSeconds(5));
```

Comment:
@awalther28 for my own education, can you elaborate on this addition?

Reply:
I think we resolved this in the split-streams PR.